### PR TITLE
feat(gltf): meshopt + texture_transform

### PR DESCRIPTION
**Pull request by belom88**
* Meshopt works good with some changes;
* Models with meshopt are generated additionally with `KHR_texture_transform` https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_texture_transform/README.md; I've done CPU transformation of UV coordinates. But textures still don't look correct.

### CODE REVIEW

File: modules/gltf/src/lib/extensions/KHR_texture_transform.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+import {Vector3, Matrix3} from '@math.gl/core';
+import type {GLTF, GLTFMaterial, GLTFMeshPrimitive} from '../types/gltf-types';
+import type {GLTFLoaderOptions} from '../../gltf-loader';
+import {BYTES, COMPONENTS} from '../api/post-process-gltf';
+import {getAccessorArrayTypeAndLength} from '../gltf-utils/gltf-utils';
+import {TypedArray} from '@loaders.gl/schema';
+
+/** Extension name */
+const EXT_MESHOPT_TRANSFORM = 'KHR_texture_transform';
```
Comment (ibgreen) [inline modules/gltf/src/lib/extensions/KHR_texture_transform.ts:13]: Typo
```suggestion
const EXT_TEXTURE_TRANSFORM = 'KHR_texture_transform';
```

File: modules/gltf/src/lib/extensions/EXT_meshopt_compression.ts [hunk 2]
```diff
@@ -16,16 +16,6 @@ const EXT_MESHOPT_COMPRESSION = 'EXT_meshopt_compression';
 
 export const name = EXT_MESHOPT_COMPRESSION;
 
-export function preprocess(gltfData: {json: GLTF}) {
-  const scenegraph = new GLTFScenegraph(gltfData);
-  if (
-    scenegraph.getRequiredExtensions().includes(EXT_MESHOPT_COMPRESSION) &&
-    !isMeshoptSupported()
```
Comment (ibgreen) [inline modules/gltf/src/lib/extensions/EXT_meshopt_compression.ts:23]: Why remove this? Not needed?

File: modules/gltf/src/lib/extensions/KHR_texture_transform.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      const matrix = makeTransformationMatrix(extensionData);
+      transformPrimitives(gltfData, i, matrix);
+    }
+  }
+}
+
+function getExtensionData(material: GLTFMaterial): any {
+  const baseColorTexture = material.pbrMetallicRoughness?.baseColorTexture;
+  if (baseColorTexture) {
+    return baseColorTexture.extensions?.[EXT_MESHOPT_TRANSFORM];
```
Comment (ibgreen) [inline modules/gltf/src/lib/extensions/KHR_texture_transform.ts:41]: Can we just pick one of these? Don't we need to apply separately to all of them? Couldn't they have different transforms?

**Review by ibgreen — COMMENTED** [review]
This looks very promising!

It would be really neat if this was split into 2 PRs, each with at least some basic unit tests.

File: modules/gltf/src/lib/extensions/EXT_meshopt_compression.ts (hunk 2 above)
Comment (belom88) [reply modules/gltf/src/lib/extensions/EXT_meshopt_compression.ts:23]: It does not work

**Pull request closed without merging**